moveit_core/trajectory_processing/src/ruckig_traj_smoothing.cpp

```diff
@@ -57,7 +57,8 @@ constexpr double MINIMUM_VELOCITY_SEARCH_MAGNITUDE = 0.01;  // rad/s. Stop searc
 
 bool RuckigSmoothing::applySmoothing(robot_trajectory::RobotTrajectory& trajectory,
                                      const double max_velocity_scaling_factor,
-                                     const double max_acceleration_scaling_factor)
+                                     const double max_acceleration_scaling_factor,
+                                     const double max_jerk_scaling_factor)
```

Comment:
Most robots are going to have velocity & accel limits either in the URDF or `joint_limits.yaml`. If there are both, then `joint_limits.yaml` overrides the URDF. So these default values aren't going to be used 90% of the time.

Here's an example joint_limits.yaml.  https://github.com/ros-planning/moveit_resources/blob/ros2/panda_moveit_config/config/joint_limits.yaml

Overall, I don't really like the approach in this PR. I think we should figure out how to get the jerk from `joint_limits.yaml` rather than using a scaling factor.

Agree with you that the user should never have to look at this file to find the defaults! We just need something in case there is no value specified elsewhere.

Reply:
Related PR:  https://github.com/ros-planning/moveit2/pull/683

Reply:
Also, I think the PR should be targeted at MoveIt/main rather than this fork, because retrieving jerk limits isn't JUST for Ruckig. It's functionality that could be used elsewhere.